orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationAiModel.java

```diff
@@ -0,0 +1,121 @@
+
+  /** Amazon Titan Text Express model */
+  public static final OrchestrationAiModel TITAN_TEXT_EXPRESS =
+      new OrchestrationAiModel("amazon--titan-text-express");
+
+  /** Azure OpenAI GPT-3.5 Turbo chat completions model */
+  public static final OrchestrationAiModel GPT_35_TURBO = new OrchestrationAiModel("gpt-35-turbo");
+
+  /** Azure OpenAI GPT-3.5 Turbo chat completions model */
+  public static final OrchestrationAiModel GPT_35_TURBO_16K =
+      new OrchestrationAiModel("gpt-35-turbo-16k");
+
+  /** Azure OpenAI GPT-4 chat completions model */
+  public static final OrchestrationAiModel GPT_4 = new OrchestrationAiModel("gpt-4");
+
+  /** Azure OpenAI GPT-4-32k chat completions model */
+  public static final OrchestrationAiModel GPT_4_32K = new OrchestrationAiModel("gpt-4-32k");
+
+  /** Azure OpenAI Text Embedding Ada 002 model */
+  public static final OrchestrationAiModel TEXT_EMBEDDING_ADA_002 =
+      new OrchestrationAiModel("text-embedding-ada-002");
+
+  /** Azure OpenAI Text Embedding 3 Small model */
+  public static final OrchestrationAiModel TEXT_EMBEDDING_3_SMALL =
+      new OrchestrationAiModel("text-embedding-3-small");
+
+  /** Azure OpenAI Text Embedding 3 Large model */
+  public static final OrchestrationAiModel TEXT_EMBEDDING_3_LARGE =
+      new OrchestrationAiModel("text-embedding-3-large");
+
+  /** Azure OpenAI GPT-4o chat completions model */
+  public static final OrchestrationAiModel GPT_4O = new OrchestrationAiModel("gpt-4o");
+
+  /** Azure OpenAI GPT-4o-mini chat completions model */
+  public static final OrchestrationAiModel GPT_4O_MINI = new OrchestrationAiModel("gpt-4o-mini");
+
+  /** Google Cloud Platform Text Bison model */
+  public static final OrchestrationAiModel TEXT_BISON = new OrchestrationAiModel("text-bison");
+
+  /** Google Cloud Platform Chat Bison model */
+  public static final OrchestrationAiModel CHAT_BISON = new OrchestrationAiModel("chat-bison");
+
+  /** Google Cloud Platform Text Embedding Gecko model */
+  public static final OrchestrationAiModel TEXT_EMBEDDING_GECKO =
+      new OrchestrationAiModel("textembedding-gecko");
+
+  /** Google Cloud Platform Text Embedding Gecko Multilingual model */
+  public static final OrchestrationAiModel TEXT_EMBEDDING_GECKO_MULTILINGUAL =
+      new OrchestrationAiModel("textembedding-gecko-multilingual");
+
+  /** Google Cloud Platform Gemini 1.0 Pro model */
+  public static final OrchestrationAiModel GEMINI_1_0_PRO =
+      new OrchestrationAiModel("gemini-1.0-pro");
+
+  /** Google Cloud Platform Gemini 1.5 Pro model */
+  public static final OrchestrationAiModel GEMINI_1_5_PRO =
+      new OrchestrationAiModel("gemini-1.5-pro");
+
+  /** Google Cloud Platform Gemini 1.5 Flash model */
+  public static final OrchestrationAiModel GEMINI_1_5_FLASH =
+      new OrchestrationAiModel("gemini-1.5-flash");
+
+  OrchestrationAiModel(@Nonnull final String modelName) {
+    setModelName(modelName);
+    setModelParams(Map.of());
+  }
```

Comment:
(Minor/Comment)

Using stateful methods inside constructor is discouraged.
Unfortunately there not many other/better options here. So we'll make an exception.